txn.go

```diff
@@ -61,13 +62,34 @@ func (d *Dgraph) NewTxn() *Txn {
 	}
 }
 
+// NewTxnInNamespace creates a new transaction in a namespace.
+// If the namespace is non-existent, the transaction will run
+// in the default namespace. Note that a Tnx created with this function
+// against an ACL-enabled Dgraph cluster will not complete successfully as the
+// namespace from the JWT is used to calculate the namespace ID.
+func (d *Dgraph) NewTxnInNamespace(namespace string) *Txn {
```

Comment:
This is how we should structure the API now:
1. NewTxn(): existing function, should create a txn in the root namespace
2. NewReadOnlyTxn(): existing function, should create a read only txn in root namespace
3. NewTxnInNamespace(): txn in a given namespace
4. NewReadOnlyTxnInNamespace(): read only txn in a given namespace

OR

1. NewTxn(TxnOptions)
2. NewReadOnlyTxn(): will deprecate this later

here, TxnOptions could be BestEffort, ReadOnly and provide namespace as well. We should update existing v2 APIs to take namespace as a TxnOption as well.

Reply:
Good call. The second way is cleaner. I added a WithNamespace to `TxnOption`s.  So it looks like:

```
txn := client.NewTxn(dgo.WithNamespace("foo"), dgo.WithReadOnly())
```

I wonder now if the namespace arg in the v2 funcs is need

Currently...
```
resp, err = client.RunDQL(ctx, dgo.RootNamespace, queryDQL, dgo.WithBestEffort())
````

Could be changed to:

```
resp, err = client.RunDQL(ctx, queryDQL, dgo.WithNamespace(dgo.RootNamespace), dgo.WithBestEffort())
```

Reply:
@aman-bansal I can make the changes to all the v2 APIs in this PR (removing the namespace attribute, replacing with the `WithNamespace` TxnOption). Just let me know

Reply:
@matthewmcneely Yes please go ahead with it.